### PR TITLE
Teardown: run the teardown fn even if the test step fails

### PR DESCRIPTION
**Pull request by wouterlucas**

### CODE REVIEW

File: src/lib/test.js [hunk 1]
```diff
@@ -74,28 +73,29 @@ const runTest = function(index) {
...
                 }
+
+                // always run test teardown
+                !!this.test.teardown
+                  ? runTeardown(this, this.test.teardown)
+                      .then(res => {
+                        done(res)
+                      })
+                      .catch(e => {
+                        done()
```
Comment (sandeep-vedam) [inline src/lib/test.js:96]: What if there is not teardown? Does it skips the teardown and ends the test if there is no teardown in the test?

File: testcases/setup-teardown-5.test.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  teardown() {
+    this.$log('I am a synchronous Teardown method that fails and is ignored')
+    throw new Error('This is a thrown error')
+  },
+  steps: [
+    {
+      description: 'The same synchronous test with assert that fails',
+      test: x => x,
+      params: 1,
+      assert: 1,
```
Comment (sandeep-vedam) [inline testcases/setup-teardown-5.test.js:16]: Here the test always pass as the params and assert are same. I think the test should be failed and then check the teardown

File: src/lib/test.js (hunk 1 above)
Comment (wouterlucas) [reply src/lib/test.js:96]: Yep, that is what !!this.test.teardown checks.

File: testcases/setup-teardown-5.test.js (hunk 2 above)
Comment (wouterlucas) [reply testcases/setup-teardown-5.test.js:16]: Correct, the description is incorrect and needs to be aligned.

**Comment by sandeep-vedam** [comment]
LGTM

**Pull request merged**